golang/cosmos/x/swingset/config.go

```diff
@@ -53,11 +73,21 @@ type SwingsetConfig struct {
 	// MaxVatsOnline is the maximum number of vats that the SwingSet kernel will have online
 	// at any given time.
 	MaxVatsOnline int `mapstructure:"max-vats-online" json:"maxVatsOnline,omitempty"`
+	// VatTranscriptRetention controls retention of vat transcript spans,
+	// and has values analogous to those of export `artifactMode` (cf.
+	// ../../../../packages/swing-store/docs/data-export.md#optional--historical-data ).
+	// * "archival": keep all transcript spans
+	// * "operational": keep only necessary transcript spans (i.e., since the
+	//   last snapshot of their vat)
+	// * "default": determined by `pruning` ("archival" if `pruning` is
+	//   "nothing", otherwise "operational")
+	VatTranscriptRetention string `mapstructure:"vat-transcript-retention" json:"vatTranscriptRetention,omitempty"`
```

Comment:
I'm not sure `omitempty` makes sense here. I'd argue this ends up required at the JS/golang interface.

Reply:
Eh, I'd rather not send an empty string only for this field if that ever comes up. If it truly were required, we'd enforce that on the JS side.

Reply:
I don't see how it could ever be empty given the resolution logic